genai.go

```diff
@@ -575,14 +689,216 @@ func (s *GenAIServiceOp) ListModels(ctx context.Context, opt *ListOptions) ([]*M
 	return root.Models, resp, nil
 }
 
-func (a Agent) String() string {
-	return Stringify(a)
+// List all knowledge bases
+func (s *GenAIServiceOp) ListKnowledgeBases(ctx context.Context, opt *ListOptions) ([]KnowledgeBase, *Response, error) {
+
+	path := KnowledgeBasePath
+	path, err := addOptions(path, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+	req, err := s.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(knowledgebasesRoot)
+	resp, err := s.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	if l := root.Links; l != nil {
+		resp.Links = l
+	}
+	if m := root.Meta; m != nil {
+		resp.Meta = m
+	}
+	return root.KnowledgeBases, resp, err
 }
 
-func (a ApiKeyInfo) String() string {
+// Create a knowledge base
+func (s *GenAIServiceOp) CreateKnowledgeBase(ctx context.Context, knowledgeBaseCreate *KnowledgeBaseCreateRequest) (*KnowledgeBase, *Response, error) {
+
+	path := KnowledgeBasePath
+
```

Comment:
For now add a check that if region is empty, to set it TOR. And also add a check that if it's anything else but TOR that it's not accepted for now. And maybe a //TODO note to remove the check if more regions get supported